modules/storage-api/src/main/java/org/apache/ignite/internal/storage/util/MvPartitionStorages.java

```diff
@@ -238,14 +240,16 @@ public CompletableFuture<Void> startRebalance(int partitionId, Function<T, Compl
                     assert old == null : createStorageInfo(partitionId);
 
                     return startRebalanceFuture;
-                }).whenComplete((unused, throwable) ->
-                        operationByPartitionId.compute(partitionId, (partId, operation) -> {
-                            assert operation instanceof StartRebalanceStorageOperation :
-                                    createStorageInfo(partitionId) + ", op=" + operation;
+                }).whenComplete((unused, throwable) -> {
+                    operationByPartitionId.compute(partitionId, (partId, operation) -> {
+                        assert operation instanceof StartRebalanceStorageOperation : createStorageInfo(partitionId) + ", op=" + operation;
 
-                            return completeOperation(operation);
-                        })
-                );
+                        return completeOperation(operation);
+                    });
+
+                    // Even if an error occurs, we must be able to abort the rebalance, so we do not report the error.
```

Comment:
Does this mean that we swallow the error?

Reply:
We don't need to handle an error on abort the rebalance, this error will be thrown to the code that triggered the rebalance.